src/mds/MDSRank.cc

```diff
@@ -901,9 +901,29 @@ Session *MDSRank::get_session(Message *m)
 {
   Session *session = static_cast<Session *>(m->get_connection()->get_priv());
   if (session) {
+    session->put(); // do not carry ref
     dout(20) << "get_session have " << session << " " << session->info.inst
 	     << " state " << session->get_state_name() << dendl;
-    session->put();  // not carry ref
+    // Check if we've imported an open session since (new sessions start closed)
+    if (session->is_closed()) {
+      Session *imported_session = sessionmap.get_session(session->info.inst.name);
+      if (imported_session && imported_session != session) {
+        dout(10) << __func__ << " replacing connection bootstrap session " << session << " with imported session " << imported_session << dendl;
+        imported_session->info.auth_name = session->info.auth_name;
+        //assert(session->info.auth_name == imported_session->info.auth_name);
```

Comment:
I presume this is for dealing with the session takeover that's being discussed for ganesha? Or is there some other way the auth names could be different?

Reply:
A race exists between a session imported from an export and an organic client connection to the rank. When the bootstrap session allcoated to the new connection is added to the session map, an existing session is found which triggers an assertion: http://tracker.ceph.com/issues/24072